### PR TITLE
Contributor ladder update

### PR DESCRIPTION
**Pull request by israel-hdez**
This is:

* Adding the `leader` and `tester` roles to the contributor ladder, which we have had for quite a long time. This is simply formalizing those roles.
  * As they are "new", I tried to describe them based on how they current behave
  * I added how to apply for these roles. This is mainly based on how to apply for the "maintainer" role. It's quite simplistic. Probably not everybody may agree but, if needed, I want to open here the discussion about if the step-in process makes sense.
* Added how to become an _Owner_ of the Kiali organization. I'm not sure if this is really right to mention.
* Updated channel to propose new contributors or changes to the governance: either by a PR or a GitHub discussion. This removes mentions to the already deprecated mail lists.
* Adding `Inactivity`, `Involuntary Removal or Demotion`, `Stepping Down/Emeritus Process` and `Voting` sections. All based on the [CNCF project template](https://github.com/cncf/project-template).
* Listing current contributors under their relevant role.

### CODE REVIEW

**Comment by israel-hdez** [comment]
I guess all people involved in Kiali project needs to either give their :+1:  or :-1: in this PR.

So, I'll ping all of you: @aljesusg, @hhovsepy, @jmazzitelli, @leandroberetta, @nrfox, @xeviknal, @abonas, @jshaughn, @lucasponce.

And, also, optionally: @mattmahoneyrh, @pbajjuri20, @prachiyadav, @skondkar.

If this is approved, I'll modify the GitHub Teams to match what is here -- mainly, remove all people who is no longer participating in Kiali.

File: GOVERNANCE.md
Comment (nrfox) [inline GOVERNANCE.md:90]: What does this mean exactly? Finding bugs in the Kiali project?

File: GOVERNANCE.md
Comment (nrfox) [inline GOVERNANCE.md:99]:
```suggestion
modifying the testers list to add the new proposed tester, or via
a [GitHub discussion](https://github.com/kiali/kiali/discussions/new) under the
```

File: GOVERNANCE.md
Comment (nrfox) [inline GOVERNANCE.md:146]:
```suggestion
Involuntary removal/demotion of a contributor happens when requirements for a certain role aren't being met. This may include repeated patterns of inactivity, extended period of inactivity, a period of failing to meet the requirements of your role, and/or a violation of the Code of Conduct. This process is important because it protects the community and its deliverables while also opens up opportunities for new contributors to step in.
```

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md:90]: I was thinking issues in the broad sense. It could be performance issues, for example. Typos are also issues, although probably typos fall under the "trivial" space...

**Comment by leandroberetta** [comment]
@israel-hdez great work!. 

A thought that I have about maintainers and testers is that, for me a tester is a mantainer too. I know you mention about the two roles not being mutually exclusive but I think it would be nice to include testers as mantainers.

**Comment by israel-hdez** [comment]
> A thought that I have about maintainers and testers is that, for me a tester is a maintainer too. I know you mention about the two roles not being mutually exclusive but I think it would be nice to include testers as maintainers.

I could agree, but then the "maintainer" role will need to be broken like this:

```
+ Maintainer
  +- Dev
  |   |- Leader
  +- Tester
```

In general, I have the perception that testers are more passive when compared to a dev. So, responsibilities are not really shared. In any case, the idea I want to transmit is that there needs to be a clear distinction between the two roles. 

I'm not sure what would be better.

File: GOVERNANCE.md
Comment (jshaughn) [inline GOVERNANCE.md:82]: Is Filip (fdrabek) also a tester?

File: GOVERNANCE.md
Comment (jshaughn) [reply GOVERNANCE.md:90]: maybe "defects" as opposed to issues?  Issues also include enhancements, at least in the github sense of 'issue'.

File: GOVERNANCE.md
Comment (jshaughn) [inline GOVERNANCE.md:166]: What if not all maintainers are available?  Can a vote be made with a quorum of maintainers, some % or minimum number of maintainers?  Can a maintainer officially abstain?  Should votes be time-boxed such that votes don't wait indefinitely for maintainers that maybe away or hard to reach?

**Review by jshaughn — COMMENTED** [review]
Great refinements.  Just a couple of comments and questions.

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md:90]: @jshaughn OK, I'll use "defets" to avoid confusions.

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md:82]: @jshaughn This Filip: https://github.com/fdrabek? I see that user is quite inactive. Also, I didn't find him on our current list of org members.

The Filip that I did find is this one: https://github.com/FilipB. In his profile, his last contribution to Kiali was June 23th. That's half the way to "inactivity". I can add him, but if he already left Kiali, I guess it's better to start clean. You say.

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md:166]: I remember this was mentioned in the discussion thread (#4199).

I thought a lot on this and I chose to left it unchanged (same policy as in the past). My current rationale is that if there is one or two maintainers unavailable, simple majority (or 2/3 vote) can still be achieved and that would mean that even if unavailable people would have voted against, that wouldn't changed the outcome.

Of course, there can be cases where several maintainers are unavailable, and that happened very recently. I had the idea that the team preference would be to wait until people become available before doing important decisions and, IMHO, this is a generous way to proceed.

About time-boxing voting, I think that would be a good idea. Perhaps limit time to 5 working days??? (which is the same limit as when proposing new maintainers).

About abstention, I would say it shouldn't be allowed. "Commitment to the project" is a pledge of a maintainer, and I think abstentions demonstrate the contrary. Anyway, perhaps can we use this PR to see how much engagement we can expect from people?? If we can't reach that 2/3 because of abstentions, we should really consider adding a paragraph about how to deal with abstentions. So far, I see some people reviewing and discussing, which is good. But I don't see any approval yet :-/

File: GOVERNANCE.md
Comment (jshaughn) [reply GOVERNANCE.md:82]: Whoops, yes, not fdrabek, Filip Brychta, not sure how I recalled Filip Frabek, that was pretty far in the past :-)  Filip Brychta is quite active in QE for Kiali, especially around performance testing.  So, sometimes contributions are not github commits.  For testers there are other ways to contribute, I think.  But there may also be a difference between project contributor and product contributor.

File: GOVERNANCE.md
Comment (jshaughn) [reply GOVERNANCE.md:166]: Yes, it's fine, we don't need to "over-engineer" the voting unless we actually end up with a problem, and then we can make an amendment.  I guess it was the "2/3 vote of all Maintainers" that made me think that all maintainers had to vote before it was official.  What you are saying here is that the voting can stop as soon as approval or rejection is mathematically secured.

Time-boxing of 5 days sounds reasonable to me.

**Comment by lucasponce** [comment]
> Added how to become an Owner of the Kiali organization. I'm not sure if this is really right to mention.

A general comment, to me, the owner/admin are more details about how Github allow to manage the users/profiles.
It's driven more for what you need to do (i.e. setup webhooks on a project, or another operation with privileges) but it shouldn't be related about the role you use (i.e. typically tester role needs privileges).

In other words, I don't think is worth to add it in a "governance" document (i.e. if we move to another repo, the governance should work as it is).

File: GOVERNANCE.md
Comment (lucasponce) [inline GOVERNANCE.md:18]: I put a general comment about "owners/admins".
I see them more like a detail depending of the task you may need, but I wouldn't add it on the file.

File: GOVERNANCE.md
Comment (lucasponce) [inline GOVERNANCE.md:50]: Discussed, the "admin/owner" is a technical detail of the github repo that can be different if the project is hosted in other tool, so I don't think it deserves this wording.
To reference to have "admin privileges" in an abstract way it should be enough (those can be granted in one or another way depending of the tool).

File: GOVERNANCE.md
Comment (lucasponce) [reply GOVERNANCE.md:166]: I think it's necessary a minor comment about when an important voting is taking place.
Like that it will give time to maintainers to attend.
To avoid situations that there is a "voting" for "tomorrow" when some folks are in PTO or any other absence.

Comment (xeviknal) [reply GOVERNANCE.md:166]: I agree with @lucasponce comment. We have to avoid the fire-fighting (again) and do things in more cruise speed.

Comment (israel-hdez) [reply GOVERNANCE.md:166]: @lucasponce @xeviknal IMHO, current wording gives this flexibility. There is no statement about how to start a vote. So, I think it's in our hands to be kind with the team and start when it's most appropriate for everybody.

However, as suggested by Jay, once the vote starts it should be time-boxed. I'm going to add this sentence in a moment.

**Comment by israel-hdez** [comment]
> I guess you mean to adjust the teams, please don't remove people, use the "contributors" group for those users that are not active.

That's fair. I think that's good to give recognition that they contributed to Kiali. What I don't like is that the "contributors" still gives Triaging privileges. My understanding is that once somebody is "removed", privileges are lost. Perhaps we should adjust the "contributors" team to remove privileges and be a place for "emeritus status".??

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md:83]: OK, I added Filip.
We can argue later if he reaches the time for "involuntary removal".

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md:50]: I think I partially agree. Decoupling from our tooling makes sense.

However, other parts already mention GitHub specific things (like GitHub discussions, issues, pull requests, teams, etc). I don't think we can fully avoid mentioning things that are specific to our tooling. So, I see that this document will anyway need to be adapted if some day we migrate to a different tool.

Anyway, I don't think we are consider moving to another tool anytime soon. So, I don't want to start thinking how to do the needed abstractions. Also, doing abstractions can add confusion so I'd lean towards being specific rather than general.

File: GOVERNANCE.md
Comment (abonas) [reply GOVERNANCE.md:167]: I would limit to more than 5 working days. Even for the reason that working days might be different from country to country (say Israel..). 7 might cover more cases and make this more inclusive.

File: GOVERNANCE.md
Comment (lucasponce) [reply GOVERNANCE.md:50]: "Admin" or "Owner" are Github terms that should not have any meaning in terms of Governance.
So they should not appear here.

Be "Maintainer" would mean to implement the privileges in the way that is required.

I'm not comfortable with those terms explictly mentioned in this document.

**Comment by lucasponce** [comment]
> That's fair. I think that's good to give recognition that they contributed to Kiali. What I don't like is that the "contributors" still gives Triaging privileges. My understanding is that once somebody is "removed", privileges are lost. Perhaps we should adjust the "contributors" team to remove privileges and be a place for "emeritus status".??

The naming and privileges can be adjusted, it was also used more for "communication" as you can tag the group and give a notification to all past members.

The idea is to not remove past contributors but change of the team, I think that's a better approach.

File: GOVERNANCE.md
Comment (jshaughn) [reply GOVERNANCE.md:167]: I am OK with the update to 5 days in the doc, although as @abonas states, 7 days may be more reasanable.  I'm resolving this conversation as I'll be fine either way.

**Review by jshaughn — DISMISSED** [review]
My questions were answered and I don't see comments from other reviewers that force me to delay an approval.  I think I'd be fine regardless of how those discussions are resolved.

**Review by aljesusg — DISMISSED** [review]
LGFM

**Review by nrfox — APPROVED** [review]

**Comment by leandroberetta** [comment]
> > A thought that I have about maintainers and testers is that, for me a tester is a maintainer too. I know you mention about the two roles not being mutually exclusive but I think it would be nice to include testers as maintainers.
> 
> @leandroberetta I could agree, but then the "maintainer" role will need to be broken like this:
> 
> ```
> + Maintainer
>   +- Dev
>   |   |- Leader
>   +- Tester
> ```
> 
> In general, I have the perception that testers are more passive when compared to a dev. So, responsibilities are not really shared. In any case, the idea I want to transmit is that there needs to be a clear distinction between the two roles.
> 
> I'm not sure what would be better.

Ok, I understand the goal, so it's fine. It was just a suggestion or thought about this.

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md]: I fixed. I removed the voting to become admin/owners.
There is already a sentence: "If needed, maintainers are elegible to become admins of specific repositories, or owners
of the Kiali organization.". I guess that's enough.

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md:156]: I modified to 7 days

**Comment by israel-hdez** [comment]
Reviewers, I added you because you are the current Kiali maintainers. So, you need to explicitly approve or reject this PR (either via a comment or via approve/request_changes) -- which is what current policy tells for governance changes. If you reject, it will be great if you can provide feedback :-)

It's been 10 days and voting is below 50%. Excluding @xeviknal who is on a leave, all others can vote.  So, I'll wait until Sept 13th 1pm CDT and **after that I will assume an approval from people who haven't emitted their vote** (excluding @xeviknal).

_Notice that for the next time,  the new policy says that voting can only last 7 working days._

BTW, that last commit dismissed your approval @nrfox, @aljesusg and @jshaughn. If you can re-review for clarity, it will be great!.

**Review by leandroberetta — APPROVED** [review]
Great work, some comments I had were asked and resolved by other team members, LGTM.

**Review by lucasponce — APPROVED** [review]
Thanks @israel-hdez for the work and the discussion.
I see all comments addressed.

**Review by nrfox — APPROVED** [review]

**Review by jshaughn — APPROVED** [review]

**Review by jmazzitelli — APPROVED** [review]
it's beautiful

File: GOVERNANCE.md
Comment (abonas) [inline GOVERNANCE.md:89]: why not settle on a single way of proposing this?

File: GOVERNANCE.md
Comment (abonas) [inline GOVERNANCE.md:127]: contribution AND communication or one of them ?

File: GOVERNANCE.md
Comment (abonas) [inline GOVERNANCE.md:134]: Demotion has some negative sound, perhaps "role change" instead?

**Review by abonas — CHANGES_REQUESTED** [review]

**Comment by abonas** [comment]
great work @israel-hdez , please see some comments/questions, it doesn't mean that this all need to be changed, in some cases clarification is sufficient. Then I'm +1 this.

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md:127]: I think it is contribution _and_ communication... 

It really depends on what is the scope of "contribution", because I understand that (for us) even ideas dropped as issue comments can be seen as contributions. If this is true, I guess that anyway you need to be totally absent to fall in inactivity.

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md:28]: Hmmm... I can see the duplicity.
My take is that the text at line 12 is the high level idea, while this list is supposed to objectively tell how to step in. So, "ability to collaborate with the team" is a little subjective IMO, but I'm not sure how we would fix this.

Even the CNCF template has it duplicated. See: https://github.com/cncf/project-template/blob/main/GOVERNANCE-maintainer.md#maintainers, so I'm not sure if this is really fixable.

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md:89]: My take is that this document exposes how to properly formalize (i.e. it's a process we cannot skip), but I don't think there should strictly be a single way to do add somebody.

I think a new member could be introduced in a meeting and if everybody gives their acknowledgement you already have reached that "simple majority voting" in advance, and we can go directly to a PR to quickly formalize (and be in compliance with governance). However, some people may not be confortable with a PR, so there is the chance of a simple discussion to also formalize.

File: GOVERNANCE.md
Comment (israel-hdez) [reply GOVERNANCE.md:134]: Isn't that covered in the next section?
I mean, I see this section mostly talking about inactive people. So, I think the whole section has a negative pitch :-/

**Comment by israel-hdez** [comment]
@abonas I wounder if yo would be OK if I fix your feedback in a follow-up PR?
I wouldn't like to see GitHub dropping other's approvals because of grammar changes :-/

**Review by abonas — APPROVED** [review]

**Comment by israel-hdez** [comment]
So, we have 7 explicit approvals out of 10 (including myself). No rejections.
It's the needed 2/3. So, I'm merging.

**Pull request merged**
